### ISSUE TITLE
As the number of collections grows CPU usage of checkhealth request also rises

### ISSUE THREAD

**Issue by jaime0815** [milvus-io/milvus#35563]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

The cluster has 450 collections.

![image](https://github.com/user-attachments/assets/42846254-e3ab-41b0-b05c-c15ef62c2063)

**Comment by xiaofan-luan** [comment]
@jaime0815 
can we do one check health on all the datanode, instead of checking everything here at datacoord?
I believe this will save many cpus

**Comment by jaime0815** [comment]
> @jaime0815 can we do one check health on all the datanode, instead of checking everything here at datacoord? I believe this will save many cpus

Yes, we also will eliminate the GetMetrics request dependency and optimize the overhead from unnecessary log.With() invocations.

**Comment by jaime0815** [comment]
A rate limiter mechanism will consume significant CPU resources as the number of collections increases, particularly due to the high CPU usage from GetMetrics requests.
